### ISSUE TITLE
Consecutive stubbing fails when using `thenThrow` at first

### ISSUE THREAD

**Issue by eder7** [mockito/mockito#1269]
Hi,

basically it's possible to change one stub's behaviour by again stubbing the same method of the same mock. Here an example:

```
when(mock.method()).thenReturn("a");
// some asserts here
when(mock.method()).thenReturn("b");
```

But when you do this...

```
when(mock.method()).thenThrow(new IllegalStateException("no no no!"));
// some asserts here
when(mock.method()).thenReturn("b");
```

...it throws the `IllegalStateException` right away during stubbing. Note that it's just a problem with `thenThrow`, not when you use `thenReturn` as the first line. This is due to mockito calling the stub answer instead of just capturing a call (to `method()`). That makes no sense in my opinion. During stubbing there should be no execution of an answer defined in a previous stubbing.

I know that it's considered bad test design. A mock's behaviour should be defined once and then it's just used. But in more complex workflows it makes absolute sense. I also know there is the possibility to define the behaviour of consecutive calls to the same method (`doThrow(...).doReturn(...)`) but it's not the same. When you have a framework around mocking tasks where you want to abstract away a layer of test setup you might end up with the need to changing the behaviour of a stub (which is actually the case here).

After all, there is no point i see (please correct me if I'm wrong) to call the stub answer during stubbing.

To reproduce you can use this code (`mockito-core:2.12.0` was used):

```
class MockitoTest {
    
    class Subject {
        fun validate() = "OK"
    }

    @Test
    fun test1() {
        val m = mock(Subject::class.java)
        `when`(m.validate()).thenThrow(IllegalStateException("x"))
    }

    @Test
    fun test2() {
        val m = mock(Subject::class.java)
        `when`(m.validate()).thenThrow(IllegalStateException("x"))
        `when`(m.validate()).thenReturn("OK")
    }

}
```

By the way: The same problem we can see at this question at Stackoverflow: https://stackoverflow.com/questions/4180209/mockito-re-stub-method-already-stubbed-with-thenthrow

The solution there was to use `reset()` but this resets *all* stubs. Not really what you'd want usually in that case...

**Comment by TimvdLippe** [comment]
This is a duplicate of #1237 and a prototype is available in #1264. Thanks for reporting though! Apparently multiple people are hitting this issue.

**Comment by eder7** [comment]
Thanks for having a look! @TimvdLippe
